core/src/test/java/com/scalar/db/storage/jdbc/JdbcTableMetadataManagerTest.java

```diff
@@ -0,0 +1,539 @@
+                new ResultSetMocker.Row(
+                    "c1", DataType.TEXT.toString(), "CLUSTERING", Order.DESC.toString(), false),
+                new ResultSetMocker.Row(
+                    "c4", DataType.BLOB.toString(), "CLUSTERING", Order.ASC.toString(), true),
+                new ResultSetMocker.Row("c2", DataType.BIGINT.toString(), null, null, false),
+                new ResultSetMocker.Row("c5", DataType.INT.toString(), null, null, false),
+                new ResultSetMocker.Row("c6", DataType.DOUBLE.toString(), null, null, false),
+                new ResultSetMocker.Row("c7", DataType.FLOAT.toString(), null, null, false)));
+    when(selectStatement.executeQuery()).thenReturn(resultSet);
+    when(connection1.prepareStatement(any())).thenReturn(selectStatement);
+
+    when(dataSource.getConnection()).thenReturn(connection1).thenReturn(connection2);
+
+    // Act
+    TableMetadata actualMetadata = manager.getTableMetadata(namespace, table);
+
+    // Assert
+    TableMetadata expectedMetadata =
+        TableMetadata.newBuilder()
+            .addPartitionKey("c3")
+            .addClusteringKey("c1", Order.DESC)
+            .addClusteringKey("c4", Order.ASC)
+            .addColumn("c1", DataType.TEXT)
+            .addColumn("c2", DataType.BIGINT)
+            .addColumn("c3", DataType.BOOLEAN)
+            .addColumn("c4", DataType.BLOB)
+            .addColumn("c5", DataType.INT)
+            .addColumn("c6", DataType.DOUBLE)
+            .addColumn("c7", DataType.FLOAT)
+            .addSecondaryIndex("c4")
+            .build();
+    assertThat(actualMetadata).isEqualTo(expectedMetadata);
+  }
+
+  @Test
+  public void deleteMetadata_forMysqlWithExistingTable_ShouldDeleteMetadata() throws SQLException {
+    deleteMetadata_ForXWithExistingTable_ShouldDeleteMetadata(
+        RdbEngine.MYSQL, "DELETE FROM `scalardb`.`metadata` WHERE `full_table_name` = 'ns1.t2'");
+  }
+
+  @Test
+  public void deleteMetadata_forPostgresqlWithExistingTable_ShouldDeleteMetadata()
+      throws SQLException {
+    deleteMetadata_ForXWithExistingTable_ShouldDeleteMetadata(
+        RdbEngine.POSTGRESQL,
+        "DELETE FROM `scalardb`.`metadata` WHERE `full_table_name` = 'ns1.t2'");
+  }
+
+  @Test
+  public void deleteMetadata_forSqlServerWithExistingTable_ShouldDeleteMetadata()
+      throws SQLException {
+    deleteMetadata_ForXWithExistingTable_ShouldDeleteMetadata(
+        RdbEngine.SQL_SERVER,
+        "DELETE FROM `scalardb`.`metadata` WHERE `full_table_name` = 'ns1.t2'");
+  }
+
+  @Test
+  public void deleteMetadata_forOracleWithExistingTable_ShouldDeleteMetadata() throws SQLException {
+    deleteMetadata_ForXWithExistingTable_ShouldDeleteMetadata(
+        RdbEngine.ORACLE, "DELETE FROM `scalardb`.`metadata` WHERE `full_table_name` = 'ns1.t2'");
+  }
+
+  private void deleteMetadata_ForXWithExistingTable_ShouldDeleteMetadata(
+      RdbEngine rdbEngine, String deleteQuery) throws SQLException {
```

Comment:
It looks like this `rdbEngine` is never used. It that okay?

Reply:
Oh thank you. Indeed it is not good as test will always be run for the same rdbengine which is not the purpose.